arangod/Agency/Supervision.cpp

```diff
@@ -1817,7 +1819,20 @@ bool arangodb::consensus::cleanupFinishedOrFailedJobsFunctional(
       }
       auto created = p.second->hasAsString("timeCreated");
       if (created) {
-        v.emplace_back(p.first, *created);
+        auto finished = p.second->hasAsString("timeFinished");
+        if (finished) {
+          try {
+            if (std::chrono::system_clock::now() -
+                    stringToTimepoint(finished.value()) >
+                std::chrono::seconds{3600}) {
+              v.emplace_back(p.first, *created);
+            }
+          } catch (...) {
+            v.emplace_back(p.first, *created);
```

Comment:
Let's add a log message at the WARN level, because this should not happen and we would be puzzled if it happens and we have no visibility.

Reply:
and TRI_ASSERT

Reply:
done